### DISCUSSION TITLE
What really happens after Apple releases a new macOS version?

### DISCUSSION THREAD

**Discussion by heyvito** [Homebrew/discussions#1667] [Everyday usage] [answered]
Hi there, folks! Hope you're all having an excellent week so far!

I've been scratching my head for a while now, and finally gathered enough courage to bug the community (that's you fine folks!) about this: What happens after Apple releases a new macOS version? Do you folks really update all formulae, and ask test-bot to bottle it? What's the timeline for it?

I'm sorry if this has been documented somewhere else, but I honestly couldn't find much material about it.

Best wishes!
– Vito

**Comment by SMillerDev** [comment, marked as answer]
> Do you folks really update all formulae, and ask test-bot to bottle it? What's the timeline for it?

Yes we do.
As for the timeline, we usually start around the final RC release.

**Reply by heyvito** [reply]
Hi Sean! Thank you so much for answering my (silly) question! ♥️ 
How interesting! Is that process fully automated? How it happens?

**Reply by SMillerDev** [reply]
Not a silly question at al.

The bottling part is fully automated (since it's essentially just a CI run). The triggering a lot less so.
One of the maintainers will usually upgrade one of the CI machines (although for 12.0 we'll have to upgrade a Intel and M1 machine) and request bottles for the big ticket items like Go, Python, etc. After that we can bottle all the things written in those languages and make an issue with the items that weren't bottled on the first try or easily fixable.

https://github.com/Homebrew/homebrew-core/issues/45514
https://github.com/Homebrew/homebrew-core/issues/64785
